README.md

```diff
@@ -52,34 +52,19 @@ interface Promise
     /**
      * Registers a callback to be invoked when the promise is resolved.
      *
-     * The callback receives `null` as first parameter and `$value` as second parameter on success. It receives the
-     * failure reason as first parameter and `null` as second parameter on failure.
-     *
      * If the promise is already resolved, the callback MUST be executed immediately.
      *
-     * Warning: If you use type declarations for `$value`, be sure to make them accept `null` in case of failures.
-     *
-     * @param callable(\Throwable|\Exception|null $exception, mixed $value) $onResolved Callback to be executed.
+     * @param callable(mixed $reason, mixed $value) <`$reason` shall be `null` on success, `$value` shall be `null` on failure>
      *
      * @return void
      */
     public function when(callable $onResolved);
 }
 ```
 
-The `when()` method MUST accept at least one argument:
-
-`$callback` – A callable conforming to the following signature:
-
-```php
-function($error, $value) { /* ... */ }
-```
-
-Any implementation MUST at least provide these two parameters. The implementation MAY extend the `Promise` interface with additional parameters passed to the callback. Further arguments to `when()` MUST have default values, so `when()` can always be called with only one argument. `when()` MAY NOT return a value. `when()` MUST NOT throw exceptions bubbling up from a callback invocation.
-
-> **NOTE:** The signature doesn't specify a type for `$error`. This is due to the new `Throwable` interface introduced in PHP 7. As this specification is PHP 5 compatible, we can use neither `Throwable` nor `Exception`.
```

Comment:
This paragraph was removed because the `$reason` is documented by docblock, further, there is no good reason to restrict implementations to use `Throwable`, `Error` or `null`; implementation should be free to choose to use a string for `$reason`.